File: Tellma/ClientApp/src/app/data/dto/get-by-id-response.ts
// tslint:disable:variable-name
import { EntityWithKey } from '../entities/base/entity-with-key';

export interface GetByIdResponse<TEntity extends EntityWithKey = EntityWithKey> {
  Extras: { [key: string]: any; };
  Result: TEntity;
  CollectionName: string;
  RelatedEntities: { [key: string]: EntityWithKey[]; };
  ServerTime: string;
}

File: Tellma/Controllers/DTO/GetByIdResponse.cs
using Tellma.Entities;
using System.Collections.Generic;
using System;

namespace Tellma.Controllers.Dto
{
    public class GetByIdResponse<TEntity> where TEntity : Entity
    {
        public Dictionary<string, object> Extras { get; set; }

        public TEntity Result { get; set; }

        public string CollectionName { get; set; }
        
        public Dictionary<string, IEnumerable<Entity>> RelatedEntities { get; set; }

        public DateTimeOffset ServerTime { get; internal set; }
    }
}
